src/tools/illink/src/linker/Linker/MessageContainer.cs

```diff
@@ -47,7 +47,7 @@ namespace Mono.Linker
 		/// <param name="subcategory">Optionally, further categorize this error</param>
 		/// <param name="origin">Filename, line, and column where the error was found</param>
 		/// <returns>New MessageContainer of 'Error' category</returns>
-		internal static MessageContainer CreateErrorMessage (string text, int code, string subcategory = MessageSubCategory.None, MessageOrigin? origin = null)
+		public static MessageContainer CreateErrorMessage (string text, int code, string subcategory = MessageSubCategory.None, MessageOrigin? origin = null)
```

Comment:
I'm curious about these - why do you need to call these? Why not just `LinkContext.LogWarning` ?

Reply:
We have a code path where we need to throw a new `LinkerFatalErrorException`.  Creating a `LinkerFatalErrorException` requires a `MessageContainer` and `CreateErrorMessage ` seemed to be the way to create a `MessageContainer`.

For more context, we have a "Copy" mode in UnityLinker where it collects all the references of a set of assemblies and copies them to an output directory.  I have this code path optimized to use a different pipeline with the bare minimum steps needed.  One of the steps I used for this mode was the [LoadReferencesStep](https://github.com/Unity-Technologies/linker/blob/unity-main/src/linker/Linker.Steps/LoadReferencesStep.cs). `LoadReferencesStep` is gone after the sync up, so I copied it to use for our "Copy" mode and that's where the `CreateErrorMessage` usage is coming from.

If you'd prefer these not be public, I can revert these.  I see that `CreateErrorMessage` isn't doing much and I can go straight to `new MessageContainer` instead.

Reply:
It's more of a "smell" for me - I consider the containers to be an internal implementation detail of the warning subsystem (errors and such are just a by product, the real meat is in the warnings). The sole reason it exists is that there's just too much information to store about a single diagnostic message, so we needed a structure around it. But the code which produces diagnostics doesn't need to know about that.

Looking at this now, personally I would prefer if you introduced something like `LinkConext.FatalError` which throws and never returns and make that public. Could be static as well I think.

But if you think it's too much refactoring I'm fine making these public as well.